core/art/views.py

```diff
@@ -1131,14 +1144,47 @@ def registerARTTest(request):
     tarindex = None
     inputfileid = None
     gitlabid = None
+    test_type = None
 
     # log all the req params for debug
     _logger.debug('[ART] registerARTtest requestParams: ' + str(request.session['requestParams']))
 
     # Checking whether params were provided
+    if 'test_type' in request.session['requestParams']:
+        test_type = request.session['requestParams']['test_type']
+
     if 'requestParams' in request.session and 'pandaid' in request.session['requestParams'] and 'testname' in request.session['requestParams']:
         pandaid = request.session['requestParams']['pandaid']
         testname = request.session['requestParams']['testname']
+    elif test_type == 'local':
+        '''
+        session_id = request.COOKIES.get('sessionid')
+        if session_id:
+        '''
+        client_ip = get_client_ip(request)
+        try:
+        # Perform reverse DNS lookup to get the client's hostname
+            art_host = socket.gethostbyaddr(client_ip)[0]
+        except socket.herror:
+        # If reverse DNS lookup fails, return the IP address as fallback
+            art_host = client_ip
+
+        if art_host.split('.')[0] in art_const.AUTHORIZED_HOSTS: 
```

Comment:
use of pass can be simplified by inverting the if statement, e.g. 
if art_host.split('.')[0] not in art_const.AUTHORIZED_HOSTS: 
    return JsonResponse({"error": "Invalid ART API user!"}, status=403)

Reply:
Removed "pass". (There were other contents here for test. I forgot to remove it in the first cleaning.).